schemas/theme/setting.json

```diff
@@ -0,0 +1,422 @@
+            }
+          }
+        }
+      }
+    },
+
+    {
+      "if": { "properties": { "type": { "const": "checkbox" } } },
+      "then": {
+        "properties": {
+          "default": {
+            "type": "boolean"
+          }
+        }
+      }
+    },
+
+    {
+      "if": { "properties": { "type": { "enum": ["number"] } } },
+      "then": {
+        "properties": {
+          "default": {
+            "type": "number"
+          },
+          "placeholder": {
+            "type": "number",
+            "description": "A placeholder value for the input."
+          }
+        }
+      }
+    },
+
+    {
+      "if": { "properties": { "type": { "enum": ["radio"] } } },
+      "then": {
+        "properties": {
+          "default": {
+            "type": "string",
+            "description": "The value of the default option"
+          },
+          "options": {
+            "$ref": "#/definitions/options"
+          }
+        },
+        "required": ["options"]
+      }
+    },
+
+    {
+      "if": { "properties": { "type": { "enum": ["range"] } } },
+      "then": {
+        "properties": {
+          "default": {
+            "type": "number"
+          },
+          "min": {
+            "type": "number",
+            "description": "The minimum value of the input"
+          },
+          "max": {
+            "type": "number",
+            "description": "The maximum value of the input"
+          },
+          "step": {
+            "type": "number",
+            "description": "The increment size between steps of the slider"
+          },
+          "unit": {
+            "type": "string",
+            "description": "The unit for the input. For example, you can set \"px\" for a font-size slider",
+            "markdownDescription": "The unit for the input. For example, you can set `px` for a font-size slider"
+          }
+        },
+        "required": ["min", "max", "step"]
+      }
+    },
+
+    {
+      "if": { "properties": { "type": { "enum": ["select"] } } },
+      "then": {
+        "properties": {
+          "default": {
+            "type": "string",
+            "description": "The value of the default option"
+          },
+          "options": {
+            "$ref": "#/definitions/options"
+          },
+          "group": {
+            "type": "string",
+            "description": "An optional attribute that you can add to each option to create option groups in the drop-down."
+          }
+        },
+        "required": ["options"]
+      }
+    },
+
+    {
+      "if": { "properties": { "type": { "enum": ["text", "textarea", "html"] } } },
+      "then": {
+        "properties": {
+          "default": {
+            "type": "string"
+          },
+          "placeholder": {
+            "type": "string",
+            "description": "A placeholder value for the input."
+          }
+        }
+      }
+    },
```

Comment:
I wonder if there's a cleaner way to do this. Because (in the current state) if you don't specify "type", then all properties for all types are shown as options on the empty object. It's not the _best_ experience. I'd much rather have only "type" be offered and then have a cascade of intellisense after.